tardis/visualization/tools/convergence_plot.py

```diff
@@ -0,0 +1,388 @@
+        """
+        Overrides default plot properties.
+
+        Parameters
+        ----------
+        fig : go.FigureWidget
+            FigureWidget object to be updated
+        parameters : dict
+            Dictionary used to update the default plot style. The dictionary should have a structure
+            like that of go.FigureWidget.to_dict(), for more information please see https://plotly.com/python/figure-structure/
+        """
+        for key, value in parameters.items():
+            if key == "data":
+                for trace in list(fig.data):
+                    self.override_plot_parameters(trace, value)
+            else:
+                if type(value) == dict:
+                    self.override_plot_parameters(fig[key], value)
+                else:
+                    fig[key] = value
+
+    def build(self, display_plot=True):
+        """
+        Creates empty plasma and luminosity convergence plots and displays them together.
+
+        Parameters
+        ----------
+        display_plot : bool
+            Displays empty plots. Defaults to True.
+        """
+        self.create_plasma_plot()
+        self.create_luminosity_plot()
+        if display_plot:
+            display(
+                widgets.VBox(
+                    [self.plasma_plot, self.luminosity_plot],
+                )
+            )
+
+    def update_plasma_plots(self):
+        """
+        Updates plasma convergence plots every iteration.
+        """
+        x = self.iterable_data["velocity"].to(u.km / u.s).value.tolist()
+        customdata = len(x) * [
+            "<br>"
+            + "Emitted Luminosity: "
+            + f'{self.value_data["Absorbed"][-1]:.2g}'
+            + "<br>"
+            + "Requested Luminosity: "
+            + f'{self.value_data["Requested"][-1]:.2g}'
+            + "<br>"
+            + "Absorbed Luminosity: "
+            + f'{self.value_data["Requested"][-1]:.2g}'
+        ]
+
+        # this adds radiation temperature vs shell velocity plot
+        self.plasma_plot.add_scatter(
+            x=x,
+            y=self.iterable_data["t_rad"],
+            line_color=self.colorscale[self.current_iteration - 1],
+            row=1,
+            col=1,
+            name=self.current_iteration,
+            legendgroup=f"group-{self.current_iteration}",
+            showlegend=False,
+            customdata=customdata,
+            hovertemplate="<b>Y</b>: %{y:.2f} at <b>X</b> = %{x:,.0f}%{customdata}",
+        )
+
+        # this adds dilution factor vs shell velocity plot
+        self.plasma_plot.add_scatter(
+            x=x,
+            y=self.iterable_data["w"],
+            line_color=self.colorscale[self.current_iteration - 1],
+            row=1,
+            col=2,
+            legendgroup=f"group-{self.current_iteration}",
+            name=self.current_iteration,
+            customdata=customdata,
+            hovertemplate="<b>Y</b>: %{y:.2f} at <b>X</b> = %{x:,.0f}%{customdata}",
+        )
+
```

Comment:
Is this where the values for Luminosity are generated for the hover box? If so, can you increase the number of significant figures output by default? 1 decimal place truncates a lot of useful info; e.g. see screenshot here. I would suggest 3 decimal places


![Screenshot 2021-06-29 at 15 34 11](https://user-images.githubusercontent.com/47385282/123816754-78c6a600-d8ef-11eb-8b1a-45711dbd630f.png)

Reply:
this applies to the other plots for T_rad too